### PR TITLE
ImagePicker component improvement

### PR DESCRIPTION
**Pull request by Momoumar**
Modify the ImagePicker Component so that the component doesn't have to save an extra copy of the picked image.
Ref issue #396  
An aia for testing: https://drive.google.com/open?id=0Bw0fmqX0oC9Yfk00Vnk4YWUzT1U2UVVHcVBNRW04TEJXcHFjUGlobnV2VWI5MkIyV3lQZ3c&authuser=0

### CODE REVIEW

File: appinventor/components/src/com/google/appinventor/components/runtime/ImagePicker.java [hunk 1]
```diff
@@ -102,114 +73,10 @@ protected Intent getIntent() {
...
-      // get the file type extension from the intent data Uri
-      ContentResolver cR = container.$context().getContentResolver();
-      MimeTypeMap mime = MimeTypeMap.getSingleton();
-      String extension = "." + mime.getExtensionFromMimeType(cR.getType(selectedImage));
-      Log.i(LOG_TAG, "extension = " + extension);
-
-      // save the image to a temp file in external storage, using a name
-      // that includes the extension
-      saveSelectedImageToExternalStorage(extension);
+      selectedImage = data.getDataString();
```
Comment (halatmit) [inline appinventor/components/src/com/google/appinventor/components/runtime/ImagePicker.java]: This code code does not work for me.   Did you test it?    I wrote a simple program where AfterPicking sets the screen  background to the result returned by the picker and it does not work.    The issue here is that data.getDataString() is a content URL and you need to return an actual file path.   The code in the version of resultReturned your removed was doing that conversion by writing the result to a file in external storage.   That seems like a waste, but I don't know a better way to do it, and I didn't see a better method on stackoverflow.   Maybe you can find it.  I might have missed something.

Let me know if you want to just give up on fixing this.  But first look arond

**Comment by halatmit** [comment]
I've been playing with this some more.  Another way to do this is to let imagePicker return the content URL.   But then you have to teach image and screenbackground and Canvas background to deal with content URLs.  This can be done.  But then I'm wondering about people who want to take the result of image picker and mail it somewhere. ..  this seems to be getting complicated.

**Comment by halatmit** [comment]
Ignore previous two comments.   I need to look at this a bit more.

**Comment by Momoumar** [comment]
@halatmit,
Yes, I tested it using the Image component (.aia file above) and it worked.
However, it seems not to be working with a screen background.

**Comment by halatmit** [comment]
@Momoumar  @jisqyv 
It turns out that what you did was correct.  The screen background problem was because a screen background won't appear unless the background color is "none".   We should fix that and I'll file a separate issue.

But it turns out that this issue I posted is more complex than I thought.   I

Briefly, the ImagePicker should return a path to a real file rather than a content URL.  The content URL is fien for setting images and screen backgrounds and the like.   But it's not good for everything.  For example, the sharing component actually needs a file.   As far as I know, the most straightforward way to deal with this is to copy the picked image to a real file, which is what happens now.

We could work on avoiding that, but it starts to get complicated.  So let's hold off.   I'm sorry to have assigned this to you -- I didn't appreciated how complicated it would become. 

Closing now, but we might come back to this some day.

Thanks for working on this.

**Pull request closed without merging**